src/Microsoft.AspNetCore.Authentication.Cookies/CookieExtensions.cs

```diff
@@ -15,7 +19,62 @@ public static class CookieExtensions
         public static IServiceCollection AddCookieAuthentication(this IServiceCollection services, Action<CookieAuthenticationOptions> configureOptions) =>
             services.AddCookieAuthentication(CookieAuthenticationDefaults.AuthenticationScheme, configureOptions);
 
-        public static IServiceCollection AddCookieAuthentication(this IServiceCollection services, string authenticationScheme, Action<CookieAuthenticationOptions> configureOptions) =>
-            services.AddScheme<CookieAuthenticationOptions, CookieAuthenticationHandler>(authenticationScheme, configureOptions);
+        public static IServiceCollection AddCookieAuthentication(this IServiceCollection services, string authenticationScheme, Action<CookieAuthenticationOptions> configureOptions)
+        {
+            // Makes sure that DataProtectionProvider is always set.
+            services.TryAddEnumerable(ServiceDescriptor.Singleton<IInitializeOptions<CookieAuthenticationOptions>, EnsureDataProtection>());
+            services.Initialize<CookieAuthenticationOptions>(authenticationScheme, options =>
```

Comment:
> Hrm, do we want to allow people to replace/customize this behavior?

Actually, that's not the scenario I had in mind. I was just wondering if making the initialization logic wouldn't prevent folks from registering the handler manually (i.e without using the extension), e.g if you don't want the `IConfiguration` importation thingy. But since the handlers are not public, this is not possible yet, even if you make the initializers public.